knora/dsplib/utils/xml_upload.py

```diff
@@ -351,42 +359,69 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
+            err=err,
+            id2iri_mapping=id2iri_mapping,
+            failed_uploads=failed_uploads,
+            stashed_xml_texts=stashed_xml_texts,
+            stashed_resptr_props=stashed_resptr_props,
+            proj_shortcode=shortcode,
+            onto_name=default_ontology,
+            server=server,
+            save_location=save_location
+        )
 
     # update the resources with the stashed XML texts
-    nonapplied_xml_texts = {}
     if len(stashed_xml_texts) > 0:
         try:
             nonapplied_xml_texts = _upload_stashed_xml_texts(verbose, id2iri_mapping, con, stashed_xml_texts)
-            nonapplied_xml_texts = _purge_stashed_xml_texts(nonapplied_xml_texts, id2iri_mapping)
+            if len(nonapplied_xml_texts) > 0:
+                raise BaseError(f"Error while trying to upload the stashed xml texts")
         except BaseException as err:
-            _handle_upload_error(err, input_file, id2iri_mapping, failed_uploads, stashed_xml_texts, stashed_resptr_props)
+            _handle_upload_error(
+                err=err,
+                id2iri_mapping=id2iri_mapping,
+                failed_uploads=failed_uploads,
+                stashed_xml_texts=stashed_xml_texts,
+                stashed_resptr_props=stashed_resptr_props,
+                proj_shortcode=shortcode,
+                onto_name=default_ontology,
+                server=server,
+                save_location=save_location
+            )
 
     # update the resources with the stashed resptrs
-    nonapplied_resptr_props = {}
     if len(stashed_resptr_props) > 0:
         try:
             nonapplied_resptr_props = _upload_stashed_resptr_props(verbose, id2iri_mapping, con, stashed_resptr_props)
-            nonapplied_resptr_props = _purge_stashed_resptr_props(stashed_resptr_props, id2iri_mapping)
+            if len(nonapplied_resptr_props) > 0:
+                raise BaseError(f"Error while trying to upload the stashed resptr props")
         except BaseException as err:
-            _handle_upload_error(err, input_file, id2iri_mapping, failed_uploads, stashed_xml_texts, stashed_resptr_props)
+            _handle_upload_error(
+                err=err,
+                id2iri_mapping=id2iri_mapping,
+                failed_uploads=failed_uploads,
+                stashed_xml_texts=stashed_xml_texts,
+                stashed_resptr_props=stashed_resptr_props,
+                proj_shortcode=shortcode,
+                onto_name=default_ontology,
+                server=server,
+                save_location=save_location
+            )
 
     # write log files
     success = True
     timestamp_str = datetime.now().strftime("%Y%m%d-%H%M%S")
-    if len(nonapplied_xml_texts) > 0:
-        _write_stashed_xml_texts(nonapplied_xml_texts, timestamp_str)
-        success = False
-    if len(nonapplied_resptr_props) > 0:
-        _write_stashed_resptr_props(nonapplied_resptr_props, timestamp_str)
-        success = False
```

Comment:
This case cannot happen any more. If one of them is > 0, this part of the code is not reached.